moziris/api/finder/pattern.py

```diff
@@ -24,38 +24,50 @@
 
 
 class Pattern:
-    """A pattern is used to associate an image file with additional attributes used in find operations.
+    """A Pattern represents a file on disk that will be used in an on-screen find operation.
 
-    While using a Region.find() operation, if only an image file is provided, Iris searches the region using a default
-    minimum similarity of 0.8. This default value can be changed in Settings.min_similarity. Using similar() you can
-    associate a specific similarity value, that will be used as the minimum value, when this pattern object is searched.
+    Iris searches a given region using the Pattern, with a default minimum similarity of 0.8.
+    This default value can be changed in Settings.min_similarity. Using similar() you can
+    change this minimum value on an instance basis when this Pattern object is searched.
     """
 
     def __init__(self, image_name: str, from_path: str = None):
+        self.caller = inspect.stack()[1][1]
+        self.temp_name = image_name
+        self.similarity = Settings.min_similarity
+        self.loaded = False
+        if from_path is not None:
+            self.load_pattern(path=from_path)
+
+    def load_pattern(self, path = None):
+        if self.loaded:
+            return
 
-        if from_path is None:
-            path = _get_image_path(inspect.stack()[1][1], image_name)
+        if path is None:
+            path = _get_image_path(self.caller, self.temp_name)
         else:
```

Comment:
Just wondering, is this `else` actually necessary?

Reply:
Yes, because this function can be invoked two different ways, resulting in `path` either being `None` or a predetermined path. If it's the former, we will look up the path on disk for the Pattern, but otherwise, we'll use the value that is passed in instead.

Reply:
Actually, no it is not. Thank you for catching that. :)